File: python/src/n0543_diameter_of_binary_tree_test.py
from n0543_diameter_of_binary_tree import diameter_of_binary_tree
from utils.tree import Tree


def test_diameter_of_binary_tree() -> None:
    tree = Tree([1, 2, 3, 4, 5])
    assert diameter_of_binary_tree(tree.root) == 3
    tree = Tree([1, 2])
    assert diameter_of_binary_tree(tree.root) == 1

File: rust/src/n0543_diameter_of_binary_tree.test.rs
use crate::{n0543_diameter_of_binary_tree::diameter_of_binary_tree, utils::tree::Tree};

#[test]
fn test_diameter_of_binary_tree() {
    let tree = Tree::new([Some(1), Some(2), Some(3), Some(4), Some(5)]);
    assert_eq!(diameter_of_binary_tree(tree.root), 3);
    let tree = Tree::new([Some(1), Some(2)]);
    assert_eq!(diameter_of_binary_tree(tree.root), 1);
}
